runtime-modules/bounty/src/benchmarking.rs

```diff
@@ -367,17 +445,164 @@ benchmarks! {
             bounty_id
         ).unwrap();
 
-        let bounty = Bounty::<T>::bounties(bounty_id);
-        assert!(matches!(
-            Bounty::<T>::get_bounty_stage(&bounty),
-            BountyStage::Withdrawal {cherry_needs_withdrawal: true, ..}
-        ));
-
-    }: withdraw_creator_funding(RawOrigin::Signed(account_id), creator.clone(), bounty_id)
+    }: withdraw_creator_cherry(RawOrigin::Signed(account_id), creator.clone(), bounty_id)
     verify {
         assert!(!Bounties::<T>::contains_key(bounty_id));
         assert_last_event::<T>(Event::<T>::BountyRemoved(bounty_id).into());
     }
+
+    announce_work_entry {
+        let cherry: BalanceOf<T> = 100.into();
+        let funding_amount: BalanceOf<T> = 100.into();
+        let (account_id, member_id) = member_funded_account::<T>("member1", 0);
+
+        let creator = BountyActor::Member(member_id);
+
+        let params = BountyCreationParameters::<T>{
+            work_period: One::one(),
+            judging_period: One::one(),
+            creator: creator.clone(),
+            cherry,
+            ..Default::default()
+        };
+
+        Bounty::<T>::create_bounty(
+            RawOrigin::Signed(account_id.clone()).into(),
+            params,
+            Vec::new()
+        ).unwrap();
+
+        let bounty_id: T::BountyId = Bounty::<T>::bounty_count().into();
+
+        assert!(Bounties::<T>::contains_key(bounty_id));
+
+        Bounty::<T>::fund_bounty(
+            RawOrigin::Signed(account_id.clone()).into(),
+            creator.clone(),
+            bounty_id,
+            funding_amount
+        ).unwrap();
+
+        let (account_id, member_id) = member_funded_account::<T>("member2", 1);
+
+    }: _(RawOrigin::Signed(account_id.clone()), member_id, bounty_id, Some(account_id.clone()))
+    verify {
+        let entry_id: T::WorkEntryId = Bounty::<T>::work_entry_count().into();
+
+        assert!(WorkEntries::<T>::contains_key(bounty_id, entry_id));
+        assert_last_event::<T>(
+            Event::<T>::WorkEntryAnnounced(bounty_id, entry_id, member_id, Some(account_id)).into()
+        );
+    }
+
+    withdraw_work_entry {
+        let cherry: BalanceOf<T> = 100.into();
+        let funding_amount: BalanceOf<T> = 100.into();
+        let (account_id, member_id) = member_funded_account::<T>("member1", 0);
+
+        let creator = BountyActor::Member(member_id);
+
+        let params = BountyCreationParameters::<T>{
```

Comment:
Shouldn't we have a stake account to take the worst case in `unlock_work_entry_stake`?

Reply:
Fixed [here](https://github.com/Joystream/joystream/pull/2218)